Src/libCZI/StreamsLib/windowsfileinputstream.cpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2023 Carl Zeiss Microscopy GmbH
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "windowsfileinputstream.h"
+
+#if defined(_WIN32)
+#include <intsafe.h>
+#include <iomanip>
+#include "../utilities.h"
+
+WindowsFileInputStream::WindowsFileInputStream(const std::string& filename)
+    : WindowsFileInputStream(Utilities::convertUtf8ToWchar_t(filename.c_str()).c_str())
+{
+}
+
+WindowsFileInputStream::WindowsFileInputStream(const wchar_t* filename)
+    : handle(INVALID_HANDLE_VALUE)
+{
+    const HANDLE h = CreateFileW(filename, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_FLAG_RANDOM_ACCESS, NULL);
+    if (h == INVALID_HANDLE_VALUE)
+    {
+        std::stringstream ss;
+        ss << "Error opening the file \"" << Utilities::convertWchar_tToUtf8(filename) << "\"";
+        throw std::runtime_error(ss.str());
+    }
+
+    this->handle = h;
+}
+
+WindowsFileInputStream::~WindowsFileInputStream()
+{
+    if (this->handle != INVALID_HANDLE_VALUE)
+    {
+        CloseHandle(this->handle);
+    }
+}
+
+void WindowsFileInputStream::Read(std::uint64_t offset, void* pv, std::uint64_t size, std::uint64_t* ptrBytesRead)
+{
+    if (size > DWORD_MAX)
+    {
+        throw std::runtime_error("size is too large");
+    }
+
+    OVERLAPPED ol = {};
+    ol.Offset = static_cast<DWORD>(offset);
+    ol.OffsetHigh = static_cast<DWORD>(offset >> 32);
+    DWORD bytes_read;
+    const BOOL B = ReadFile(this->handle, pv, static_cast<DWORD>(size), &bytes_read, &ol);
```

Comment:
Either we can explicitly remove the B variable here and put ReadFile into if clause or add a more descriptive name like fileReadSuccessful

Reply:
your wish is my command, done, thanks